client/battle/BattleFieldController.cpp

```diff
@@ -333,6 +358,205 @@ std::set<BattleHex> BattleFieldController::getHighlightedHexesForMovementTarget(
+	const CStack * hoveredStack = owner.curInt->cb->battleGetStackByPos(hoveredHex, true);
+	if(!(hoveredStack && hoveredStack->isShooter()))
+		return rangedFullDamageLimitHexes;
+
+	auto rangedFullDamageDistance = GameConstants::BATTLE_PENALTY_DISTANCE;
+
+	// overwrite full ranged damage distance from Additional info field of LIMITED_SHOOTING_RANGE bonus
+	auto bonus = hoveredStack->getBonus(Selector::type()(BonusType::LIMITED_SHOOTING_RANGE));
+	if(bonus != nullptr && bonus->additionalInfo != CAddInfo::NONE)
+		rangedFullDamageDistance = bonus->additionalInfo[0];
+
+	// from ranged full damage hexes get only the ones at the limit
+	for(auto & hex : rangedFullDamageHexes)
+	{
+		if(BattleHex::getDistance(hoveredHex, hex) == rangedFullDamageDistance)
+			rangedFullDamageLimitHexes.push_back(hex);
+	}
+
+	return rangedFullDamageLimitHexes;
+}
+
+std::vector<std::vector<BattleHex::EDir>> BattleFieldController::getOutsideNeighbourDirectionsForLimitHexes(std::vector<BattleHex> rangedFullDamageHexes, std::vector<BattleHex> rangedFullDamageLimitHexes)
+{
+	std::vector<std::vector<BattleHex::EDir>> output;
+
+	if(rangedFullDamageHexes.empty())
+		return output;
+
+	for(auto & hex : rangedFullDamageLimitHexes)
+	{
+		// get all neighbours and their directions
+		auto neighbours = hex.neighbouringTilesWithDirection();
+
+		std::vector<BattleHex::EDir> outsideNeighbourDirections;
+
+		// for each neighbour add to output only the ones not found in rangedFullDamageHexes
+		for(auto & neighbour : neighbours)
+		{
+			auto it = std::find(rangedFullDamageHexes.begin(), rangedFullDamageHexes.end(), neighbour.second);
+
+			if(it == rangedFullDamageHexes.end())
+				outsideNeighbourDirections.push_back(neighbour.first); // push direction
+		}
+
+		output.push_back(outsideNeighbourDirections);
+	}
+
+	return output;
+}
+
+// krs: This needs to be moved somewhere else
+unsigned char setBit(unsigned char mask, int position)
+{
+    if (position >= 0 && position < 8)
+        mask |= (1 << position);
+
+    return mask;
+}
+
+std::vector<std::shared_ptr<IImage>> BattleFieldController::calculateFullRangedDamageHighlightImages(std::vector<std::vector<BattleHex::EDir>> fullRangedDamageLimitHexesNeighbourDirections)
+{
+	std::vector<std::shared_ptr<IImage>> output; // if no image is to be shown an empty image is still added to help with traverssing the range
+
+	if(fullRangedDamageLimitHexesNeighbourDirections.empty())
+		return output;
+
+	for(auto & directions : fullRangedDamageLimitHexesNeighbourDirections)
+	{
+		std::shared_ptr<IImage> highlightImage;
+		uint8_t mask = 0;
+		
+		// convert directions to mask
+		for(auto direction : directions)
+			mask = setBit(mask, direction);
+
+		// mask will have 1 on the bits corresponding to edge positon 0..5
+		//    /\
+		//   0  1
+		//  /    \
+		// |      |
+		// 5      2
+		// |      |
+		//  \    /
+		//   4  3
+		//    \/
+		switch (mask)
+		{
+		// single edges
+		case 0b000001: // top left only

```

Comment:
I would name your images accordingly: hexEdge000001.png, ...000010.png, 001110.png, etc

Reply:
I have chosen these long names for multiple edges, because I found it too difficult to differentiate only from binary mask :(.
If you are not strongly against I would keep verbose names.

Reply:
Use `std::map<int, std::shared_ptr<IImage>` instead, with mask from here used as key.
Would also remove need for ~20 members in class.

Reply:
Also - on naming:
IMO human-readable names are OK, but perhaps unify it a bit? You have images named 0.png ... 5.png alongside with human-readable names.

Reply:
Implemented with maps